modules/storages/app/services/storages/one_drive_managed_folder_sync_service.rb

```diff
@@ -126,6 +126,12 @@ def create_folder(project_storage, folder_path)
         .call(storage: @storage, folder_path:)
         .match(on_failure: ->(error) { format_and_log_error(error, folder_path:) },
                on_success: ->(folder_info) do
+                 last_project_folder = ::Storages::LastProjectFolder
+                                         .find_by(
+                                           project_storage_id: project_storage.id,
+                                           mode: project_storage.project_folder_mode
+                                         )
+                 last_project_folder.update(origin_folder_id: folder_info.id)
                  project_storage.update(project_folder_id: folder_info.id)
```

Comment:
🟡 This should be an atomic operation (meaning both writes should succeed) so we don't end up with weird results.

I'd chain the calls if we don't care about one of the operations (`project_storage.update and last_project_folder.update`) or wrap this in a transaction if we want to make sure that both succeed and fail together.

Reply:
so, question: option a or b? chain two separate calls? or introduce transaction for writing this to db?

Reply:
:+1: I suppose to be correct we should use transaction around both operations. I will add it.

Reply:
@mereghost @Kharonus it has been done. Have a look, please.